### PR TITLE
diff: add unit test for chunk split (#209)

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 
1. add unit test
2. fix make, upgrade to go 1.12

### What is changed and how it works?
git cherry pick #209 

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -58,9 +60,7 @@ fmt:
 check:
 	#go get github.com/golang/lint/golint
 	@echo "vet"
-	@ go tool vet $(FILES) 2>&1 | awk '{print} END{if(NR>0) {exit 1}}'
-	@echo "vet --shadow"
-	@ go tool vet --shadow $(FILES) 2>&1 | awk '{print} END{if(NR>0) {exit 1}}'
+	#$(GO) vet -all $(PACKAGES) 2>&1 | tee /dev/stderr | $(FAIL_ON_STDOUT)
```
Comment (WangXiangUSTC) [inline Makefile]: this line will exit, don't know why, uncomment after fix it

**Review by csuzhangxc — COMMENTED** [review]
LGTM

**Comment by WangXiangUSTC** [comment]
@kennytm PTAL

**Review by amyangfei — APPROVED** [review]
LGTM

**Pull request merged**